### PR TITLE
Armor Expression

### PR DESCRIPTION
**Pull request by frackz**
### Description
I made a expression so instead of you typing all types of armor to create a list of all player's armor, its just an expression called player's armor. I tested it ingame, and it works just fine :)

---
**Target Minecraft Versions:** any
**Requirements:** i guess none
**Related Issues:** https://github.com/SkriptLang/Skript/issues/4752

### CODE REVIEW

**Comment by TheBentoBox** [comment]
Couldn't this be made to work for any living entity using [`EntityEquipment`](https://purpurmc.org/javadoc/org/bukkit/inventory/EntityEquipment.html) instead of `PlayerEquipment`?

File: src/main/java/ch/njol/skript/expressions/ExprGetAllArmor.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+   private Expression<Player> player;
+ 
+   @Override
+   public Class<? extends ItemStack> getReturnType() {
+       return ItemStack.class;
+   }
+ 
+   @Override
+   public boolean isSingle() {
+       return true;
```
Comment (TheLimeGlass) [inline src/main/java/ch/njol/skript/expressions/ExprGetAllArmor.java:48]: When you always return multiple values like your array list there. The isSingle should be false.

**Review by TheLimeGlass — CHANGES_REQUESTED** [review]
It should also support multiple values and use getArray(event), but if you're switching over to a property expression like stated above, then the property expression will help in that regard.

**Comment by dkfrede** [comment]
> Couldn't this be made to work for any living entity using [`EntityEquipment`](https://purpurmc.org/javadoc/org/bukkit/inventory/EntityEquipment.html) instead of `PlayerEquipment`?

So, i have some problems trying your idea, i can still get the player's armor but not a skeleton's or a mobs.

**Comment by dkfrede** [comment]
> This should also use `SimplePropertyExpression` instead of `SimpleExpression` Take a look at [this expression](https://github.com/SkriptLang/Skript/blob/master/src/main/java/ch/njol/skript/expressions/ExprAI.java) for example

This should work.

**Comment by TheLimeGlass** [comment]
The return type doesn't need to be an array and there is a register method in the class to use instead.

**Comment by dkfrede** [comment]
> The return type doesn't need to be an array and there is a register method in the class to use instead.

If you can suggest the changes, it would be great, im going down to eat :)

**Comment by dkfrede** [comment]
> The return type doesn't need to be an array and there is a register method in the class to use instead.
Do you know the problem " i can still get the player's armor but not a skeleton's or a mobs."

File: src/main/java/ch/njol/skript/expressions/ExprGetAllArmor.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import org.eclipse.jdt.annotation.Nullable;
+ 
+import ch.njol.skript.Skript;
+import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import ch.njol.skript.lang.ExpressionType;
+
+public class ExprGetAllArmor extends SimplePropertyExpression<LivingEntity, ItemStack[]> {
+ 
+   static {
+       Skript.registerExpression(ExprGetAllArmor.class, ItemStack[].class, ExpressionType.COMBINED, "%livingentities%'s armor");
```
[the remark this answers is not in the document]
Comment (AyhamAl-Ali) [reply src/main/java/ch/njol/skript/expressions/ExprGetAllArmor.java]: Please don't mark reviews as resolved if they're not

**Review by AyhamAl-Ali — CHANGES_REQUESTED** [review]
_Follow up review_

This expression should also support **change methods** (change, acceptChange) I would suggest **SET** & **DELETE** modes

**Comment by TheLimeGlass** [comment]
@frackz why was this branch deleted? Mistake? Or you used your master for something else?

**Comment by TheLimeGlass** [comment]
Appears author has deleted and abandoned this pull request. Closing in favor of https://github.com/SkriptLang/Skript/pull/5456

**Pull request closed without merging**